From 2e584940a76e6d4074882358d90ad56ad3cf6094 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 21 Dec 2022 07:34:24 +0200
Subject: [PATCH] [SW-113135] gaudi3: remove rst_src fixup for pldm

On latest pldm images, reset source value reflects correctly
as cold reset upon de-assert of pldm (h9_rst_n), hence removing
reset source fixup added for pldm.

Change-Id: I889f26979d935b7c800730a8972cb89b3e87eb98
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 7ff38b1..766d2fc 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -256,13 +256,11 @@
 		preboot_cfg_dma_mmu_bypass(false);
 }
 
-__weak u32 gaudi3_rst_src_fixup(u32 rst_src) { return rst_src; }
-
 u32 hl_rst_src_get(void)
 {
 	u32 rst_src = regmap_readl(mmD0_PSOC_GLOBAL_CONF_RST_SRC);
 
-	return gaudi3_rst_src_fixup(rst_src);
+	return rst_src;
 }
 
 void init_prstn(void)
diff --git a/arch/gaudi3/arc/hl.h b/arch/gaudi3/arc/hl.h
index 67c8057..fe87240 100644
--- a/arch/gaudi3/arc/hl.h
+++ b/arch/gaudi3/arc/hl.h
@@ -57,7 +57,6 @@
 u32 hl_get_die_id(void);
 bool hl_is_dual_die_sku(void);
 u64 gaudi3_dma_addr_fixup(u64 loadaddr);
-u32 gaudi3_rst_src_fixup(u32 rst_src);
 void pif_arc_pcie_monitoring(void);
 
 #endif /* ARCH_GAUDI3_ARC_HL_H_ */
diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index ae7efbf..7fb5f83 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -133,21 +133,6 @@
 	}
 }
 
-/*
- * In paladium emulator the full design reset signal sets the reset
- * cause to "manual reset" when in fact the entire design is being
- * reset same as in cold reset of the asic. In order to properly support PCI
- * init that depends on the reset source, the "manual reset" source in
- * paladium is treated as cold reset
- */
-u32 gaudi3_rst_src_fixup(u32 rst_src)
-{
-	if (rst_src == HL_MANUAL_RST)
-		return HL_COLD_RST;
-
-	return rst_src;
-}
-
 void hl_dev_cpu_pll_init(void)
 {
 	hl_pll_init(gaudi3_cpu_pll_cfg, 1);
